### ISSUE TITLE
[Bug]: Expand all rows table header button shows the wrong aria-label

### ISSUE THREAD

**Issue by jdcai** [carbon-design-system/carbon#10254]
### Package

carbon-components-react

### Browser

Chrome

### Package version

v7.49.1

### Description

When all the rows are collapsed the header expand all button's aria-label shows "Collapse all rows" instead of "Expand all rows"
![image](https://user-images.githubusercontent.com/4312305/145254645-00ec3219-1c37-450b-8831-4417a64277d3.png)

And then the opposite happens here
![image](https://user-images.githubusercontent.com/4312305/145254702-9711d1f3-20b4-41a3-8649-4915d2308829.png)

Fix should be to just change !isExpanded to isExpanded here
https://github.com/carbon-design-system/carbon/blob/83f414f2c9186796bd31959786369c903eb959af/packages/react/src/components/DataTable/DataTable.js#L259

### CodeSandbox example

https://react.carbondesignsystem.com/?path=/story/components-datatable-expansion--batch-expansion

### Steps to reproduce

https://react.carbondesignsystem.com/?path=/story/components-datatable-expansion--batch-expansion

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
